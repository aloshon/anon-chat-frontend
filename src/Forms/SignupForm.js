import React, {useContext} from "react";
import {useHistory} from "react-router-dom";
import useFields from '../Hooks/useFields';
import "./Forms.css";
import UserContext from "../UserContext";
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';

const SignupForm = () => {
    const history = useHistory();
    const {signup} = useContext(UserContext);
    const [formData, handleChange] = useFields({
        username: '',
        password: ''
    });

    const handleSubmit = async (e) => {
        e.preventDefault();
        try{
            let res = await signup(formData);
            if(res){
                history.push('/');
            }
        }catch(e){
            console.log(`Error signing up!:${e}`)
            alert(e)
        }
    }

    return (
        <Container className="postform">
            <h1>Sign Up</h1>
            <Form onSubmit={handleSubmit}>
                <Form.Group>
                    <Form.Label>Username:</Form.Label>
                    <Form.Control name="username" maxLength="30" type="text" value={formData.username} onChange={handleChange} placeholder="Username" />
                </Form.Group>
                <Form.Group>
                    <Form.Label>Password:</Form.Label>
                    <Form.Control name="password" type="password" value={formData.password} onChange={handleChange} placeholder="Password" />
                </Form.Group>
                <Button variant="primary" type="submit" size="md" block>
                    Submit
                </Button>
            </Form>
        </Container>
    )
}

export default SignupForm;