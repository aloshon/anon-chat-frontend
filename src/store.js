import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);