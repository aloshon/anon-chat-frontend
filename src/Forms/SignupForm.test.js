import { render, screen } from '@testing-library/react';
import SignupForm from './SignupForm';
import { Provider } from "react-redux";
import { store } from "../store"
import {BrowserRouter} from 'react-router-dom';
import UserContext from "../UserContext";


describe("SignupForm should render form for user to sign up", () => {
    const signup = () => {
        const user = {
            id: 1,
            username: "user1",
            isAdmin: false,
            blockList: []
        }
        return user
    }

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <UserContext.Provider value={{signup}}>
                <SignupForm />
            </UserContext.Provider>
        </BrowserRouter>
      </Provider>
    );
    test("renders labels", () => {
      const label1 = screen.getByText("Username:");
      const label2 = screen.getByText("Password:");
      const button = screen.getByText("Submit");
      expect(label1).toBeInTheDocument();
      expect(label2).toBeInTheDocument();
      expect(button).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });