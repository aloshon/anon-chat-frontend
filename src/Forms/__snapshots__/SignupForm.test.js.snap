// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm should render form for user to sign up matches snapshot 1`] = `<DocumentFragment />`;
